pkg/collector/corechecks/cluster/orchestrator/transformers/k8s/horizontalpodautoscaler.go

```diff
@@ -0,0 +1,421 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build orchestrator
+
+package k8s
+
+import (
+	v2 "k8s.io/api/autoscaling/v2"
+
+	model "github.com/DataDog/agent-payload/v5/process"
+
+	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/cluster/orchestrator/transformers"
+)
+
+// ExtractHorizontalPodAutoscaler returns the protobuf model corresponding to a Kubernetes Horizontal Pod Autoscaler resource.
+// https://github.com/kubernetes/api/blob/v0.23.15/autoscaling/v2/types.go#L33
+func ExtractHorizontalPodAutoscaler(v *v2.HorizontalPodAutoscaler) *model.HorizontalPodAutoscaler {
+	if v == nil {
+		return &model.HorizontalPodAutoscaler{}
+	}
+
+	m := &model.HorizontalPodAutoscaler{
+		Metadata: extractMetadata(&v.ObjectMeta),
+		Spec:     extractHorizontalPodAutoscalerSpec(&v.Spec),
+		Status:   extractHorizontalPodAutoscalerStatus(&v.Status),
+	}
+
+	if len(v.Status.Conditions) > 0 {
+		hpaConditions, conditionTags := extractHorizontalPodAutoscalerConditions(v)
+		m.Conditions = hpaConditions
+		m.Tags = append(m.Tags, conditionTags...)
+	}
+
+	m.Tags = append(m.Tags, transformers.RetrieveUnifiedServiceTags(v.ObjectMeta.Labels)...)
+	return m
+}
+
+// extractHorizontalPodAutoscalerSpec converts Kubernetes HorizontalPodAutoscalerSpec to our custom one
+// https://github.com/kubernetes/api/blob/v0.23.15/autoscaling/v2/types.go#L51
+func extractHorizontalPodAutoscalerSpec(s *v2.HorizontalPodAutoscalerSpec) *model.HorizontalPodAutoscalerSpec {
+	spec := model.HorizontalPodAutoscalerSpec{
+		Target: &model.HorizontalPodAutoscalerTarget{
+			Kind: s.ScaleTargetRef.Kind,
+			Name: s.ScaleTargetRef.Name,
```

Comment:
I feel like we miss api version https://github.com/kubernetes/api/blob/v0.23.15/autoscaling/v2/types.go#L92

Reply:
👍 indeed, but not a blocker as kind/name should give enough context even though we miss the version